p2p/conn/connection_test.go

```diff
@@ -769,3 +770,273 @@ func stopAll(t *testing.T, stoppers ...stopper) func() {
+		messagingRate     time.Duration // rate at which messages are sent
+		totalDuration     time.Duration // total duration for which messages are sent
+		sendQueueCapacity int           // send queue capacity i.e., the number of messages that can be buffered
+		sendRate          int64         // send rate in bytes per second
+		recRate           int64         // receive rate in bytes per second
+	}{
+		{
+			// testcase 1
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// SendQueueCapacity should not affect transmission delay.
+			name: "queue capacity = 1, " +
+				"total load = 50 KB, " +
+				"msg rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 1,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 2
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// Increasing SendQueueCapacity should not affect transmission
+			// delay.
+			name: "queue capacity = 50, " +
+				"total load = 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 50,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 3
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// Increasing SendQueueCapacity should not affect transmission
+			// delay.
+			name: "queue capacity = 100, " +
+				"total load = 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 100,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 4
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// The test runs for 100 messages, expecting a total transmission time of ~2 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 2 * 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          2 * 50,
+			sendQueueCapacity: 10,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 5
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// The test runs for 400 messages,
+			// expecting a total transmission time of ~8 seconds.
```

Comment:
[question] is there a way to add the expected ~8 seconds as a verification step of the test case? If not, does the developer running the test look at some test output and verify that it's 8 seconds?

Reply:
Cool thanks! Can resolve.